### ISSUE TITLE
bug: sorting hex

### ISSUE THREAD

**Issue by yjkimjunior** [paritytech/Nomidot#163]
Currently storing BigNumbers as hexadecimal as prisma can only handle up to 53 bit integers, but the thing is that we cannot sort hexadecimals (i don't think?).

Maybe there is a way, but we certainly won't get anything reliable 'order_by' on these. 

Not sure yet how to fix...

**Comment by amaurymartiny** [comment]
> but the thing is that we cannot sort hexadecimals (i don't think?)

yeah, unless we write a SCALE-encoded compact integer converter as a psql plugin, i don't think we can sort.

postgresql handles up to u64, so if the runtime exposes u128 (like the current balance), then for lossless storage, our only way is to store hex strings/binary.

we _could_ for now store them as u64 though, and wait for the day it breaks (which could happen in 5 months or 5 years or 50 years), but i'm willing to try that.

**Comment by yjkimjunior** [comment]
> yeah, unless we write a SCALE-encoded compact integer converter as a psql plugin

have you done this before (do you think this would be realistic for us to do)?
